src/gui/FxMixerView.cpp

```diff
@@ -432,53 +444,39 @@ void FxMixerView::deleteUnusedChannels()
 
 
 
-void FxMixerView::moveChannelLeft(int index)
+void FxMixerView::moveChannelLeft(int index, int focusIndex)
 {
 	// can't move master or first channel left or last channel right
 	if( index <= 1 || index >= m_fxChannelViews.size() ) return;
 
-	int selIndex = m_currentFxLine->channelIndex();
+	FxMixer *m = Engine::fxMixer();
 
-	FxMixer * mix = Engine::fxMixer();
-	mix->moveChannelLeft(index);
+	// Move instruments channels
+	m->moveChannelLeft( index );
 
-	// refresh the two mixer views
-	for( int i = index-1; i <= index; ++i )
-	{
-		// delete the mixer view
-		int replaceIndex = chLayout->indexOf(m_fxChannelViews[i]->m_fxLine);
+	// Update widgets models
+	m_fxChannelViews[index]->setChannelIndex( index - 1 );
+	m_fxChannelViews[index - 1]->setChannelIndex( index );
 
-		chLayout->removeWidget(m_fxChannelViews[i]->m_fxLine);
-		m_racksLayout->removeWidget( m_fxChannelViews[i]->m_rackView );
-		delete m_fxChannelViews[i]->m_fader;
-		delete m_fxChannelViews[i]->m_muteBtn;
-		delete m_fxChannelViews[i]->m_soloBtn;
-		delete m_fxChannelViews[i]->m_fxLine;
-		delete m_fxChannelViews[i];
+	// Swap positions in array
+	qSwap(m->m_fxChannels[index], m->m_fxChannels[index - 1]);
 
-		// add it again
-		m_fxChannelViews[i] = new FxChannelView( m_channelAreaWidget, this, i );
-		chLayout->insertWidget( replaceIndex, m_fxChannelViews[i]->m_fxLine );
-		m_racksLayout->insertWidget( replaceIndex, m_fxChannelViews[i]->m_rackView );
-	}
+	// Focus on new position
+	setCurrentFxLine( focusIndex );
```

Comment:
What is the point of this? If I move a channel, that doesn't imply I want the moved channel necessarily selected...